packages/immutadot-parser/src/parser.utils.ts

```diff
@@ -0,0 +1,95 @@
+import { NavType } from "./enums";
+
+const mapIfNotNull = <T, R> (maybe: T, fn: (v: T) => R): R => maybe === null ? null : fn(maybe);
```

Comment:
This could be helpful https://github.com/gcanti/fp-ts ?

Reply:
Seems nice but I'd rather avoid dependencies